backend/elastic.js

```diff
@@ -113,6 +140,28 @@ class Elastic {
    * @param  {integer} max    The index of last document to retreive
    */
   async search(query, termId, min, max) {
+    this.subjects = this.subjects || new Set(await this.getSubjectsFromClasses());
```

Comment:
I know this works and everything, but do you think its a good idea to just use an `if` statement instead? Lot clearer for people reading the code and a lot easier to understand (even for people who have a lot of experience). 

Up to you :)

Reply:
Oh totally, yeah I've been seeing that pattern a lot lately and got used to it but I think you're right. Just changed it.